service/raft/raft_group0.hh

```diff
@@ -110,11 +101,31 @@ public:
     // it can't do it by itself (it's dead).
     future<> leave_group0(std::optional<gms::inet_address> host = {});
 
+private:
     void init_rpc_verbs();
     future<> uninit_rpc_verbs();
 
     // Handle peer_exchange RPC
     future<group0_peer_exchange> peer_exchange(discovery::peer_list peers);
+
+    raft_server_for_group create_server_for_group(raft::group_id id, raft::server_address my_addr);
+
+    // Loads server address for group 0 from disk if present, otherwise randomly generates a new one and persists it.
+    // Execute on shard 0 only.
+    future<raft::server_address> load_or_create_my_addr();
+
+    // Run the discovery algorithm.
+    //
+    // Discovers an existing group 0 cluster or elects a server (called a 'leader')
+    // responsible for creating a new group 0 cluster if one doesn't exist
+    // (in particular, we may become that leader).
+    //
+    // See 'raft-in-scylla.md', 'Establishing group 0 in a fresh cluster'.
+    future<group0_info> discover_group0(raft::server_address my_addr);
+
+    // Called by `discover_group0`.
+    // Precondition: `_group0` contains `persistent_discovery`.
+    future<group0_info> do_discover_group0(raft::server_address my_addr);
```

Comment:
This comment doesn't explain why we have two discovers, both private. do_discover_group0 obviously is a bad name since it doesn't reflect how this fucntion is different from discover_group0. The comment doesn't tell me either. Going forward, when I maintain this code, I won't know what to put where. Most likely I'll just do something arbitrary eventually making these both functions a spaghetti.

Reply:
> run_discovery could probably even become a member function of persistent_discovery at this point.

Of course we would have to rename some variables inside the implementation of `run_discovery`, but the logic itself is not dependant on group 0 in any way.

Reply:
Moving it into persistent_discovery is a good idea.